src/main/java/com/cognifide/aemrules/checks/ContentResourceCheck.java

```diff
@@ -0,0 +1,237 @@
+import org.sonar.plugins.java.api.tree.Tree.Kind;
+import org.sonar.plugins.java.api.tree.VariableTree;
+
+@Rule(
+    key = ContentResourceCheck.RULE_KEY,
+    name = ContentResourceCheck.RULE_MESSAGE,
+    priority = Priority.MINOR,
+    tags = Tags.AEM
+)
+public class ContentResourceCheck extends BaseTreeVisitor implements JavaFileScanner {
+
+  public static final String RULE_KEY = "AEM-18";
+
+  public static final String RULE_MESSAGE = "Always null check the returned value of Page.getContentResource() method";
+
+  public static final String METHOD_NAME = "getContentResource";
+
+  public static final String NOT_EQUAL = "NOT_EQUAL_TO";
+
+  public static final String EQUAL = "EQUAL_TO";
+
+  private boolean returnOccurredInsideEqualNullCheck = false;
+
+  private boolean insideEqualNullCheck = false;
+
+  private JavaFileScannerContext context;
+
+  private Map<String, Boolean> contentResources = new HashMap();
+
+  private Set<String> resourcesToIgnore = new HashSet<>();
+
+  @Override
+  public void scanFile(JavaFileScannerContext context) {
+    this.context = context;
+    scan(context.getTree());
+  }
+
+  @Override
+  public void visitBinaryExpression(BinaryExpressionTree tree) {
+    if (NOT_EQUAL.equals(tree.kind().name())) {
+      if (isResourceNullChecked(tree.leftOperand(), tree.rightOperand())) {
+        contentResources.replace(tree.leftOperand().firstToken().text(), true);
+      } else if (isResourceNullChecked(tree.rightOperand(), tree.leftOperand())) {
+        contentResources.replace(tree.rightOperand().firstToken().text(), true);
+      }
+    }
+    super.visitBinaryExpression(tree);
+  }
+
+  @Override
+  public void visitIfStatement(IfStatementTree tree) {
+    if (isThisAResourceNullCheck(tree, NOT_EQUAL)) {
+      // Resource name can be either on left or right side of the if statement
+      if (isRightSideNullCheck(tree)) {
+        contentResources.replace(tree.condition().firstToken().text(), true);
+      } else {
+        contentResources.replace(tree.condition().lastToken().text(), true);
+      }
+    } else if (isThisAResourceNullCheck(tree, EQUAL)) {
+      insideEqualNullCheck = true;
+    }
+    super.visitIfStatement(tree);
+    if (returnOccurredInsideEqualNullCheck) {
+      if (isRightSideNullCheck(tree)) {
+        resourcesToIgnore.add(tree.condition().firstToken().text());
+      } else {
+        resourcesToIgnore.add(tree.condition().lastToken().text());
+      }
+    }
+    // cleaning flags after if statement
+    if (isRightSideNullCheck(tree)) {
+      contentResources.replace(tree.condition().firstToken().text(), false);
+
+    } else {
+      contentResources.replace(tree.condition().lastToken().text(), false);
+    }
+    insideEqualNullCheck = false;
+  }
+
+  @Override
+  public void visitAssignmentExpression(AssignmentExpressionTree tree) {
+    if (Constants.RESOURCE_TYPE.equals(tree.variable().symbolType().fullyQualifiedName()) && tree
```

Comment:
This if statement is hard to read. Please refactor it to more meaningful methods